### ISSUE TITLE
Polymer 1.x

### ISSUE THREAD

**Issue by mhrisse** [ebidel/polymer-gmail#6]
@ebidel Excellent example. Any plans to upgrade this to use v1.x?

**Comment by ebidel** [comment]
Yea. Theres a WIP branch.

**Comment by mhrisse** [comment]
Thanks!

**Comment by devAtPolydeals** [comment]
@ebidel I downloaded the V1.X build and get ``` Uncaught SyntaxError: Unexpected reserved word```
at https://github.com/ebidel/polymer-gmail/blob/1.0/scripts/app.js#L17'
Any idea on how to fix this?

**Comment by ebidel** [comment]
You'll need to `npm install` and run babel to compile the es6 code. Namely, the `import` is the thing causing the problem. When I move the 1.0 branch to default, I'll publish some instructions on how to get up and running. I would have liked to keep the app free from compilation, but ES6 classes were just so nice :)

**Comment by devAtPolydeals** [comment]
@ebidel Ok Thanks. I figured there was some kinda new age sorcery going on in there which i didn't know about :~) BTW it works well in the 'dist' directory after gulp however there is one small bug. In dist\index.html the script tag still looks for ```<script src="scripts/app.js"></script>``` however in ```bundlejs``` gulp task it ends up creating bundle.js in the dist\script folder. You may have to change either gulp or index.

**Comment by ebidel** [comment]
Hmm, it shouldn't: https://github.com/ebidel/polymer-gmail/blob/1.0/index.html#L199. I would kill and unregister any service worker in `chrome://serviceworker-internals/` and try re-running. You may also have to do that for dev'ing vs. serving out of dist/

**Comment by ebidel** [comment]
**Update**: https://poly-mail.appspot.com/ now serves the Polymer 1.0 version! I did a quick comparison on WPT to see the different from the Polymer 0.5 version of the app.

*TL;DR*: **1.0 version reduced first paint by 4s and lowered speed index by 52%!**

First view - *Motorola G - Chrome - 3G connection*

  | 0.5 app ([results](http://www.webpagetest.org/result/150710_3R_19D3/))  | 1.0 app ([results](http://www.webpagetest.org/result/150722_HW_1940/))
------------- | ------------- | -------------
Speed index | 13564  | 7133
First paint  | 6.743s  | 2.783s
First byte | 13.443s | 1.918s

First view - *Desktop Chrome - cable connection*

 | 1.0 app ([results](http://www.webpagetest.org/result/150722_F6_1BPF/))
------------- | ------------- 
Speed index | 1675 
First paint  | 0.694s  
First byte | 1.008s 


**Notes**
- The 1.0 version has offline caching via `localStorage` and Service worker, however this should not play into the first view metrics listed above.
- The 1.0 version uses [lazy loading](https://gist.github.com/ebidel/1ba71473d687d0567bd3) to load the webcomponents.js polyfills and async html imports. It would be nice to switch back to the way the 0.5 app loaded these resources to see more of a 1:1 comparison. If I have time, I'll do that and update the table above.

**Comment by ragingwind** [comment]
:tada: Great job!

**Comment by devAtPolydeals** [comment]
@ebidel  sorry its the other way round, index.html points to bundle.js at https://github.com/ebidel/polymer-gmail/blob/1.0/index.html#L199 but the scripts folder has app.js https://github.com/ebidel/polymer-gmail/tree/1.0/scripts; ofcourse this issue does not occur in dist folder because of browserify. However i mostly use app folder for studying the code so always run from app in chrome dev tools.

**Comment by devAtPolydeals** [comment]
Tried running the new https://poly-mail.appspot.com/ , sign in went through fine but emails did not load; got the below error in console.


![image](https://cloud.githubusercontent.com/assets/12582040/8839533/36e4aace-308b-11e5-97e4-0fac72b4c557.png)

**Comment by ebidel** [comment]
@devAtPolydeals I don't have a good solution for running it under dev without compiling atm. app.js will need to be [compiled](https://github.com/ebidel/polymer-gmail/blob/1.0/gulpfile.js#L170) to bundle.js and the app run from dist/. I need to setup watch tasks to get things working in dev.

That error is probably around https://github.com/ebidel/polymer-gmail/blob/1.0/scripts/googleapis.js#L124-L127. Something is different about some of your email threads. Would love to know more.

**Comment by devAtPolydeals** [comment]
@ebidel  I would love to help but don't know how? It is tough to debug it with minified js and I cannot run uniminified on localhost because of googleapi does not allow access from localhost. 

BTW ?debug switch gives 404 because users.json is not available. 

```Failed to load resource: the server responded with a status of 404 (Not Found)           https://poly-mail.appspot.com/data/users.json ```

**Comment by strycore** [comment]
@devAtPolydeals try replacing the Google client ID with your own (which must have GMail read access and optionally G+ read access). 
I also was encountering an error (not sure it was the same as you) until I replaced the client ID with my own, the app ran fine after that.

**Comment by ebidel** [comment]
@devAtPolydeals, ack on users.json. It's a known issue. I need to upload a list of folks...that aren't my actual contacts :p

**Comment by ebidel** [comment]
Merged the code into master. Closing this.
